docs/src/app/(public)/(content)/react/handbook/customization/page.mdx

```diff
@@ -0,0 +1,124 @@
+# Customization
+
+<Subtitle>A guide to customizing the behavior of Base UI components.</Subtitle>
+<Meta name="description" content="A guide to customizing the behavior of Base UI components." />
+
+## Base UI events
+
+Change events such as `onOpenChange`, `onValueChange`, and `onPressedChange` are custom to Base UI.
+They can be emitted by various different DOM events, effects, or even during rendering.
+
+```js title="Base UI event signatures"
+onOpenChange: (open, eventDetails) => void
+onValueChange: (value, eventDetails) => void
+onPressedChange: (pressed, eventDetails) => void
+```
+
+The `eventDetails` property is passed as a second argument to Base UI event handlers.
+This enables the event to be customized, and also allows you to conditionally run side effects based on the reason or DOM event that caused the change.
+
+```tsx title="eventDetails object"
+interface BaseUIEventDetails {
+  reason: string;
+  event: Event;
```

Comment:
I don't know about this. Two things that feel a bit off:

- Under React JS terminology, this would be a native event

``` suggestion
  nativeEvent: Event;
```

- React has a bunch of logic to normalize event behavior https://github.com/facebook/react/blob/e02c173fa597ce914c6876f71556c39bf20f40dc/packages/react-dom-bindings/src/events/SyntheticEvent.js#L333. Not exposing it could deprive users of important behavior.

Reply:
Many times the events are called from a native event, so we used native events everywhere so it's consistent no matter how the event handlers are registered. 

If people request React events, it could be added as a `.reactEvent` property (that's potentially undefined). These change events are different from React event handlers to begin with signature-wise etc., so it's not a big deal I think

Reply:
@atomiks In the end, I was wondering about having:
```suggestion
    event?: React.SyntheticEvent;
    nativeEvent: Event;
```

but yeah, nothing critical, it's more of a v2 kind of potential change, depending on how people respond.